### PR TITLE
Test: integration test of the Data Server

### LINKED ISSUE #646: [TEST] Data Server Integration
### What do you want to test?
Full integration test for data server.

### Outline of test structure
Should use all components typically found in the data server and run without error, but not using launchpad. See the jax IPPO example for how the components are typically logically distributed between core nodes. If multiple tests are required to cover different distinct cases (e.g. value-based vs policy-gradient), then they should be included.

### Definition of done
Test(s) passing and including all typical components.

### Mandatory checklist before making a PR
* [ ] The success criteria laid down in “Definition of done” are met.
* [ ] Test code is documented - docstrings for methods and classes, static types for arguments.
* [ ] Documentation is updated - README, CONTRIBUTING, or other documentation.

### PR DESCRIPTION
**Pull request by OmaymaMahjoub**
## What?
Dataserver integration test
## Why?
Part of on going efforts to increase testing code coverage of callback redesigned Mava.
## Extra
Close #646 
The data_server tested only in the case of single process

### CODE REVIEW

File: tests/jax/integration/data_server_test.py [hunk 1]
```diff
@@ -12,91 +12,134 @@
...
-            )
+    table_trainer = data_server.server_info()["trainer"]
+    assert table_trainer.num_episodes == 5  # 5 episodes
+    assert table_trainer.num_unique_samples != 0
 
-        # On policy params
-        assert_if_value_is_not_none(
-            system_config.get("max_queue_size"), table_info.max_size
-        )
+    # dataset added by the exevutor via the adders
```
Comment (siddarthsingh1) [inline tests/jax/integration/data_server_test.py:99]: `exevutor` should be `executor`.

**Review by siddarthsingh1 — DISMISSED** [review]
Looks good. Thanks @OmaymaMahjoub. There's just a small typo in one of the comments. I also wanted to ask, we only test the use of the `TrajectoryDataset` since we only have IPPO, would we need to add the `TransitionDataset` after DQN is available?

**Comment by OmaymaMahjoub** [comment]
> Looks good. Thanks @OmaymaMahjoub. There's just a small typo in one of the comments. I also wanted to ask, we only test the use of the `TrajectoryDataset` since we only have IPPO, would we need to add the `TransitionDataset` after DQN is available?

Thanks @siddarthsingh1  for the review! We made the two types testing in the unit test, and I believe that the integration test needs to be on a system that is predefined  which is in our cas IPPO. However, if you feel the importance of making `TransitionDataset` integration test I will try to see how to manage and add it to a customized system

File: tests/jax/integration/data_server_test.py (hunk 1 above)
Comment (OmaymaMahjoub) [reply tests/jax/integration/data_server_test.py]: Done!

**Review by siddarthsingh1 — DISMISSED** [review]
Thanks for the explanation @OmaymaMahjoub. The current test seems fine 👍

File: tests/jax/integration/data_server_test.py [hunk 2]
```diff
@@ -12,91 +12,134 @@
...
+from tensorflow.python.data.ops import dataset_ops
+
+from mava.systems.jax import System
+from tests.jax.integration.mock_systems import mock_system_single_process
+
+
+@pytest.fixture
+def test_system() -> System:
+    """A built IPPO system"""
+    return mock_system_single_process()
```
Comment (DriesSmit) [inline tests/jax/integration/data_server_test.py:29]: Thanks, @OmaymaMahjoub :slightly_smiling_face:  Looks good. Are we only testing multi_process for the integration tests?

Comment (OmaymaMahjoub) [reply tests/jax/integration/data_server_test.py:29]: Thanks @DriesSmit  for reviewing! Unfortunately since data server a Reverb node it was tough to find out how to test it in the multi process so at least we tested it in the single process case

Comment (DriesSmit) [reply tests/jax/integration/data_server_test.py:29]: That makes sense.

**Review by DriesSmit — APPROVED** [review]
Looks good, thanks @OmaymaMahjoub :fire:

File: tests/jax/integration/data_server_test_data.py [hunk 3]
```diff
@@ -1,100 +0,0 @@
-# python3
```
Comment (AsadJeewa) [inline tests/jax/integration/data_server_test_data.py:1]: happy that we have removed this

**Review by AsadJeewa — APPROVED** [review]
Extensive testing. Thanks for all your efforts @OmaymaMahjoub

File: tests/jax/integration/data_server_test_data.py (hunk 3 above)
Comment (OmaymaMahjoub) [reply tests/jax/integration/data_server_test_data.py:1]: It was before attached to an old test was made (draft)
